### ISSUE TITLE
Remove data within specified timestamp range?

### ISSUE THREAD

**Issue by bartmika** [nakabonne/tstorage#18]
Hello @nakabonne, great job with the library! I have been looking through the code and am unable to find out how to delete data within specified timestamp range. Is this feature implemented? 

The closest I was able to find was inside [storage_examples_test.go#L129](https://github.com/nakabonne/tstorage/blob/main/storage_examples_test.go#L129) where you use a `os.RemoveAll(tmpDir)`. 

Thank you and appreciate your response!

**Comment by nakabonne** [comment]
Hi @bartmika thank you for proposing and such a kind word! As you mentioned the only way to delete data is removing directories manually.

However, I'm looking to support a new method like `Delete(start, end int64)` (I didn't open an issue as I don't need it soon)

I think it should remove data in partitions within the specified range. Let's say for example. You give this range.

```go
_ := storage.Delete(1600000002, 1600000010)
```

It will delete only the second partition.

```
  │      ┌───────────────────┐ max: 1600000010
  ├──Ignore──>  Partition
  │      └───────────────────┘ min: 1600000009
  │
  │      ┌───────────────────┐ max: 1600000008
  ├──Delete──>  Partition
  │      └───────────────────┘ min: 1600000005
  │
  │      ┌───────────────────┐ max: 1600000004
  └──Ignore──>  Partition
         └───────────────────┘ min: 1600000000
```

**Comment by wattanakorn495** [comment]
I really need this feature.
I am looking for a retention rule feature that automatically removes the old partition.

**Comment by nakabonne** [comment]
@wattanakorn495 Right. We may want some kind of option like `WithRetentionDuration`.

```go
	storage, err := tstorage.NewStorage(
		tstorage.WithDataPath("./data"),
		tstorage.WithRetentionDuration(time.Hour*24)
	)
```

I'm planning to support it as well as `Delete(start, end int64)`.

**Comment by nakabonne** [comment]
Both are cool proposals. I really appreciate you!

**Comment by hitoshi44** [comment]
Hello @nakabonne. I read your awesome article about this repository with a lot of fun.
I think I can help this `Delete`  stuff, but before do that, I'd like to make some points more claer out.

### Does `Delete` delete only by `Partision`s, or by `DataPoint`s ?
If an answer to this question is "No" (thus `Delete` can delete by `DataPoint`s.), then next question appears.

### `Delete` should distinguish `Metric` and `Labels` ? 
Because `Partition` has `Metric` and `Labels` for manage data points, I think `Delete` also  should distinguish them. So `Delete` can be defined either like below

- `Delete(from, to Row) error` 
- `Detete(from, to int64, met Metric, labels []Labels)`

Using `Row` to specify from and to point, or using from-to as timestamp(int64) and taking Metric and []Labels to determine which metric to delete.

Or like the example in thid thread you put, delete points in range in all metrics. 

- `Delete(from, to int64) error`

**Comment by bartmika** [comment]
Hi @nakabonne just wanted to give this issue a *poke* due to @hitoshi44 recent response as this would be an awesome feature to have! Thank you and hope you had a great New Year/holidays.
